diff --git a/src/cmakeFindModules/OpenMVGConfig.cmake.in b/src/cmakeFindModules/OpenMVGConfig.cmake.in
index 01d4df98..64a4739d 100644
--- a/src/cmakeFindModules/OpenMVGConfig.cmake.in
+++ b/src/cmakeFindModules/OpenMVGConfig.cmake.in
@@ -42,7 +42,7 @@ MACRO(OPENMVG_REPORT_NOT_FOUND REASON_MSG)
   # FindPackage() use the camelcase library name, not uppercase.
   IF (OPENMVG_FIND_QUIETLY)
     MESSAGE(STATUS "Failed to find OPENMVG - " ${REASON_MSG} ${ARGN})
-  ELSE (OPENMVG_FIND_REQUIRED)
+  ELSEIF (OPENMVG_FIND_REQUIRED)
     MESSAGE(FATAL_ERROR "Failed to find OPENMVG - " ${REASON_MSG} ${ARGN})
   ELSE()
     # Neither QUIETLY nor REQUIRED, use SEND_ERROR which emits an error
