Index: cmake-3.16.3/Tests/CMakeLists.txt
===================================================================
--- cmake-3.16.3.orig/Tests/CMakeLists.txt
+++ cmake-3.16.3/Tests/CMakeLists.txt
@@ -1491,6 +1491,7 @@ ${CMake_SOURCE_DIR}/Utilities/Release/pu
     )
   list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/ExternalProject")
   set_tests_properties(ExternalProject PROPERTIES
+    ENVIRONMENT GIT_ALLOW_PROTOCOL=file
     RUN_SERIAL 1
     TIMEOUT ${CMAKE_LONG_TEST_TIMEOUT})
 
@@ -2557,6 +2558,7 @@ ${CMake_SOURCE_DIR}/Utilities/Release/pu
         -P "${CMake_BINARY_DIR}/Tests/CTestUpdateGIT.cmake"
         )
       list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/${CTestUpdateGIT_DIR}")
+      set_property(TEST CTest.UpdateGIT PROPERTY ENVIRONMENT GIT_ALLOW_PROTOCOL=file)
     endif()
 
     # Test CTest Update with HG
