Index: cmake-3.16.3/Tests/CMakeLib/testUVProcessChain.cxx
===================================================================
--- cmake-3.16.3.orig/Tests/CMakeLib/testUVProcessChain.cxx
+++ cmake-3.16.3/Tests/CMakeLib/testUVProcessChain.cxx
@@ -181,6 +181,10 @@ bool checkOutput(std::istream& outputStr
   }
 
   std::string error = getInput(errorStream);
+  auto qemu_error_pos = error.find("qemu:");
+  if (qemu_error_pos != std::string::npos) {
+    error.resize(qemu_error_pos);
+  }
   if (error.length() != 3 || error.find('1') == std::string::npos ||
       error.find('2') == std::string::npos ||
       error.find('3') == std::string::npos) {
