Index: cmake-3.16.3/Source/kwsys/testSystemTools.cxx
===================================================================
--- cmake-3.16.3.orig/Source/kwsys/testSystemTools.cxx
+++ cmake-3.16.3/Source/kwsys/testSystemTools.cxx
@@ -328,7 +328,15 @@ static bool CheckFileOperations()
   }
 
   // While we're at it, check proper TestFileAccess functionality.
-  if (kwsys::SystemTools::TestFileAccess(testNewFile,
+  bool do_write_test = true;
+#if defined(__linux__)
+  // If we are running as root on linux ignore this check, as
+  // root can always write to files
+  do_write_test = (getuid() != 0);
+#endif
+
+  if (do_write_test &&
+      kwsys::SystemTools::TestFileAccess(testNewFile,
                                          kwsys::TEST_FILE_WRITE)) {
     std::cerr
       << "TestFileAccess incorrectly indicated that this is a writable file:"
