Index: android-platform-external-boringssl/linux-arm/crypto/fipsmodule/bsaes-armv7.S
===================================================================
--- android-platform-external-boringssl.orig/linux-arm/crypto/fipsmodule/bsaes-armv7.S
+++ android-platform-external-boringssl/linux-arm/crypto/fipsmodule/bsaes-armv7.S
@@ -92,7 +92,7 @@
 _bsaes_decrypt8:
 	adr	r6,_bsaes_decrypt8
 	vldmia	r4!, {q9}		@ round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	r6,.LM0ISR
 #else
 	add	r6,r6,#.LM0ISR-_bsaes_decrypt8
@@ -583,7 +583,7 @@ _bsaes_const:
 _bsaes_encrypt8:
 	adr	r6,_bsaes_encrypt8
 	vldmia	r4!, {q9}		@ round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	r6,.LM0SR
 #else
 	sub	r6,r6,#_bsaes_encrypt8-.LM0SR
@@ -1018,7 +1018,7 @@ _bsaes_encrypt8_bitslice:
 _bsaes_key_convert:
 	adr	r6,_bsaes_key_convert
 	vld1.8	{q7},  [r4]!		@ load round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	r6,.LM0
 #else
 	sub	r6,r6,#_bsaes_key_convert-.LM0
Index: android-platform-external-boringssl/src/crypto/fipsmodule/aes/asm/bsaes-armv7.pl
===================================================================
--- android-platform-external-boringssl.orig/src/crypto/fipsmodule/aes/asm/bsaes-armv7.pl
+++ android-platform-external-boringssl/src/crypto/fipsmodule/aes/asm/bsaes-armv7.pl
@@ -746,7 +746,7 @@ $code.=<<___;
 _bsaes_decrypt8:
 	adr	$const,_bsaes_decrypt8
 	vldmia	$key!, {@XMM[9]}		@ round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	$const,.LM0ISR
 #else
 	add	$const,$const,#.LM0ISR-_bsaes_decrypt8
@@ -845,7 +845,7 @@ _bsaes_const:
 _bsaes_encrypt8:
 	adr	$const,_bsaes_encrypt8
 	vldmia	$key!, {@XMM[9]}		@ round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	$const,.LM0SR
 #else
 	sub	$const,$const,#_bsaes_encrypt8-.LM0SR
@@ -953,7 +953,7 @@ $code.=<<___;
 _bsaes_key_convert:
 	adr	$const,_bsaes_key_convert
 	vld1.8	{@XMM[7]},  [$inp]!		@ load round 0 key
-#ifdef	__APPLE__
+#if defined(__thumb2__) || defined(__APPLE__)
 	adr	$const,.LM0
 #else
 	sub	$const,$const,#_bsaes_key_convert-.LM0
